src/functional/ffront/field_operator_ast.py

```diff
@@ -176,6 +158,14 @@ class Return(Stmt):
 
 class FieldOperator(LocatedNode, SymbolTableTrait):
     id: SymbolName  # noqa: A003
-    params: list[DataSymbol]
+    params: list[Symbol]
     body: list[Stmt]
     closure: list[Symbol]
+
+    @validator("params")
+    def validate_params_type(cls, params: list[Symbol]):
+        assert all(
+            isinstance(param.type, (common_types.DeferredSymbolType, common_types.DataType))
+            for param in params
+        )
+        return params
```

Comment:
Could this custom validator be replaced by a generic definition of the `Symbo` node (as `Symbol[T]`)?